### ISSUE TITLE
v2.4.3 build failure on macOS 11 for Homebrew

### ISSUE THREAD

**Issue by p-linnane** [OpenPrinting/cups#715]
**Describe the bug**
Compilation of cups v2.4.3 fails on macOS 11, but succeeds on macOS 12 and macOS 13

**To Reproduce**
1. Download the v2.4.3 tarball
2. `./configure --disable-debug --disable-dependency-tracking --prefix=/usr/local/Cellar/cups/2.4.3 --libdir=/usr/local/Cellar/cups/2.4.3/lib --with-components=core --without-bundledir`
3. `make install`
4. Error:
```
  Archiving libcups.a...
  Undefined symbols for architecture x86_64:
    "_SecTrustCopyCertificateChain", referenced from:
        _httpCopyCredentials in tls.o
  ld: symbol(s) not found for architecture x86_64
  clang: error: linker command failed with exit code 1 (use -v to see invocation)
  make[1]: *** [libcups.2.dylib] Error 1
```

**Expected behavior**
cups compiles

**Screenshots**
N/A

**System Information:**
 - OS and its version: macOS 11 x86_64 & macOS 11 arm64
 - Application N/A
 - CUPS version 2.4.3

**Additional context**
Relates to https://github.com/Homebrew/homebrew-core/pull/132550

**Comment by zdohnal** [comment]
Hi,

interesting, neither of mentioned functions we have in our codebase - only the similar functions without '_'. Maybe brew adds underscores to them or it happens in xcode related files?

@michaelrsweet I've checked tls-darwin.c/.h and I don't see any MacOS relevant headers, only function prototypes in tls-darwin.h - are those functions linked during runtime?

**Comment by carlocab** [comment]
> interesting, neither of mentioned functions we have in our codebase - only the similar functions without '_'. Maybe brew adds underscores to them or it happens in xcode related files?

On macOS, symbols are prefixed with `_`. You can ignore the `_` for the purposes of debugging this.

**Comment by michaelrsweet** [comment]
@carlocab It doesn't surprise me, and this changed in 9f04d166bd to stop using a deprecated API. And this for the TLS support code on macOS that is itself going away in CUPS 2.5/3.0...

I guess the question is whether we want to revert this change...

**Comment by zdohnal** [comment]
@michaelrsweet I would propose to move the commit into 2.5 and properly announce it in CHANGES.md, revert it for 2.4.x, WDYT?

**Comment by michaelrsweet** [comment]
@zdohnal I will be pulling all of the tls-darwin.* code in 2.5.x (along with the Windows tls-sspi.* code) so any fix/reversion should only happen in the 2.4.x branch.
